zonemaster/zonemaster-backend
# Parse config

## Purpose

This PR adds Zonemaster::Backend::Config->parse(). This method makes unit testing more predictable by letting unit tests embed their configuration(s) instead of having Zonemaster::Backend::Config figure out which file to open and read and risking to use another configuration than the one the developer intended.

An illustration of how such unit tests can be written is provided in t/config.t.

## Context

This PR is made as a first step towards #685, but it also provides a way to update other unit tests to make them more predictable.

## Changes

Zonemaster::Backend::Config:
* Z::B::Config->parse() is broken out from Z::B::Config->load_config().
* Existing validations are moved out from accessor methods and into parse().
* Some drive-by clean-up of the Z::B::Config code.
* A seconds-to-minutes conversion is moved away from Z::B::Config and into business logic code.

Dependencies:
* Add test dependency on Test::NoWarnings. Used in the new t/config.t. It makes unit tests fail if warnings are emitted.
* Install libio-stringy-perl on Debian. It's an undeclared dependency of libconfig-inifiles-perl.

Travis config:
* The version number of to Perl 5.32 is fixed. (Credits to @pnax for investigating.)

## How to test this PR

Run the unit tests. They're automatically run as part of CI but in principle it's a good idea to verify they also work locally.

The new t/config.t file contains tests for all important aspects of Z::B::Config->parse().
The old t/test01.t verifies that the old Z::B::Config->load_config() still works the same way.
